packages/tx-construction/src/createTransactionInternals.ts

```diff
@@ -14,34 +15,56 @@ export type CreateTxInternalsProps = {
   requiredExtraSignatures?: Crypto.Ed25519KeyHashHex[];
 };
 
-export const createTransactionInternals = async ({
+/** Returns transaction body based on the information available before input selection takes place */
+export const createPreInputSelectionTxBody = ({
   auxiliaryData,
   withdrawals,
   certificates,
   validityInterval,
-  inputSelection,
   collaterals,
   mint,
   scriptIntegrityHash,
-  requiredExtraSignatures
-}: CreateTxInternalsProps): Promise<Cardano.TxBodyWithHash> => {
+  requiredExtraSignatures,
+  outputs
+}: Omit<CreateTxInternalsProps, 'inputSelection'> & { outputs?: Cardano.TxOut[] }): {
+  body: TxBodyPreInputSelection;
+  auxiliaryData?: Cardano.AuxiliaryData;
+} => {
+  const body: TxBodyPreInputSelection = {
+    auxiliaryDataHash: auxiliaryData ? Cardano.computeAuxiliaryDataHash(auxiliaryData) : undefined,
+    certificates,
+    mint,
+    outputs: outputs || [],
+    requiredExtraSignatures,
+    scriptIntegrityHash,
+    validityInterval,
+    ...(withdrawals?.length && { withdrawals }),
+    ...(collaterals && { collaterals: [...collaterals] })
+  };
+
+  return {
+    auxiliaryData,
+    body
+  };
+};
+
+/** Updates the txBody after input selection takes place with the calculated change and selected inputs */
+export const includeChangeAndInputs = async ({
```

Comment:
This is declared as an `async function`, but it seems to be sync... am I missing something?

Reply:
Done here: [b1a7f4d](https://github.com/input-output-hk/cardano-js-sdk/pull/1099/commits/b1a7f4de95cf58257f625ea9f49fc9f385d25382)